File: 10-armstrong-check/armstrong.rb
# Ruby program to check if number is armstrong
# Written by M.V.Harish Kumar on 26-01-2025

print "Enter the number: "
num = gets.chomp

arm_num = 0
num.chars.each do |c|
  arm_num += c.to_i ** num.length
end

puts "#{num} is #{'NOT ' if num.to_i != arm_num}a Armstrong number"

File: 10-armstrong-check/armstrong.go
// Go program to check if a given number is armstrong or not
// Written by M.V.Harish Kumar on 18-06-2025
package main

import (
	"fmt"
	"math"
)

func main() {
	var num, armNum uint;
	fmt.Print("Enter a number: ")
	fmt.Scanln(&num)

	numLen := math.Floor(math.Log10(float64(num))) + 1
	armNum = 0
	tmp := num

	for tmp != 0 {
		armNum += uint(math.Pow(float64(tmp % 10), numLen))
		tmp /= 10
	}

	if num == armNum {
		fmt.Println(num, "is an Armstrong number")
	} else {
		fmt.Println(num, "is NOT an Armstrong number")
	}
}
